py3status/modules/gazelle.py

```diff
@@ -0,0 +1,153 @@
+                data={'username': self.username, 'password': self.password},
+                headers={'User-Agent': 'Mozilla/5.0'},
+                cookiejar=self._cookiejar
+                )
+        try:
+            self._index()  # just to try if login info are good
+        except RequestException:
+            self.py3.log(
+                    "Cannot login to %s" % self.baseurl,
+                    level=self.py3.LOG_ERROR
+                    )
+            raise
+        self._cookiejar.save(self.baseurl)
+        self.py3.log(
+                "Saved cookie in storage",
+                level=self.py3.LOG_INFO
+                )
+
+    def _index(self):
+        resp = self.py3.request(
+                self.baseurl+"/ajax.php",
+                params={'action': "index"},
+                headers={
+                    'User-Agent': 'Mozilla/5.0'
+                    },
+                cookiejar=self._cookiejar
+                )
+        return resp.json()
+
+    def post_config_hook(self):
+        self._cookiejar = self.py3.storage_new_cookiejar()
+        try:
+            self._cookiejar.load(self.baseurl)
+            self._index()  # just to try if saved cookie is good
+        except RequestException:
+            self.py3.log(
+                    "Cookie invalid, outdated or missing. Re-logging.",
+                    level=self.py3.LOG_INFO
+                    )
+            self._login()
+
+    def gazelle(self):
+        index = self._index()
+        if index['status'] != "success":
+            msg = "Tracker failed to give me user info"
+            self.py3.log(msg, level=self.py3.LOG_ERROR)
+            raise RequestException(msg)
+        result = index['response']
+        data = {
+            'ratio': result['userstats']['ratio'],
+            'requiredratio': result['userstats']['requiredratio'],
+            'download': self._readable_size(result['userstats']['downloaded']),
+            'upload': self._readable_size(result['userstats']['uploaded']),
+            'messages': result['notifications']['messages'],
+            'notifs': result['notifications']['notifications']
+            }
+
+        # Ok, so if you think that what I'll do next stinks a bit, remember that
+        # the color threshold design in py3status is a gigantic turd.
+        # Main reason of this ugliness is the impossibility to natively define a
+        # variable threshold, so I have to replace keyword by the actual value.
+        for n, item in enumerate(self.thresholds['ratio']):
+            if item[0] == "requiredratio":
+                lst = list(item)
+                lst[0] = data['requiredratio']
+                self.thresholds['ratio'][n] = tuple(lst)
+        # End of horrible code
+
+        self.py3.threshold_get_color(data['ratio'], 'ratio')
+        self.py3.threshold_get_color(data['messages'], 'messages')
+        self.py3.threshold_get_color(data['notifs'], 'notifs')
```

Comment:
(Q) Is it more useful to use `urgent` instead of thresholds for `messages` and/or `notifs`?

Reply:
dunno, maybe.

Reply:
I'll set 2 new boolean variables `urgent_new messages` and `urgent_new_notifs`